supervisor/supervisor.go

```diff
@@ -48,9 +48,6 @@ func New(stateDir string, runtimeName, shimName string, runtimeArgs []string, ti
 	}
 	go s.exitHandler()
 	go s.oomHandler()
-	if err := s.restore(); err != nil {
```

Comment:
@hqhq the healthcheck is a different service and is registered beforehand: https://github.com/containerd/containerd/pull/1177/files#diff-f138749ec16f01ba07c3faa14dfc5234R216

My only issue with this order is that if the restore gets deadlocked somehow, the request will stay deadlocked forever. And docker doesn't use a timeout on its contexts atm. I'd rather the requests fail with a "no such service" error that causing a deadlock.